### PR TITLE
WebDriver: Add computedrole endpoint

### PR DESCRIPTION
**Pull request by Epigenetic**
This adds a missing endpoint in #15551.

### CODE REVIEW

File: Userland/Libraries/LibWeb/WebDriver/Client.cpp [hunk 1]
```diff
@@ -72,6 +72,7 @@ static constexpr auto s_webdriver_endpoints = Array {
     ROUTE(POST, "/session/:session_id/window/fullscreen"sv, fullscreen_window),
     ROUTE(POST, "/session/:session_id/element"sv, find_element),
     ROUTE(POST, "/session/:session_id/elements"sv, find_elements),
+    ROUTE(GET, "/session/:session_id/element/:element_id/computedrole"sv, get_computed_role),
```
Comment (trflynn89) [inline Userland/Libraries/LibWeb/WebDriver/Client.cpp:75]: So far every method declaration and definition is in spec order. I'd really like to keep it that way. So this belongs just below the `is_element_enabled` line.

File: Userland/Libraries/LibWeb/WebDriver/Client.h [hunk 2]
```diff
@@ -75,6 +75,7 @@ class Client : public Core::Object {
     virtual Response get_element_rect(Parameters parameters, JsonValue payload) = 0;
     virtual Response is_element_enabled(Parameters parameters, JsonValue payload) = 0;
     virtual Response click(Parameters parameters, JsonValue payload) = 0;
+    virtual Response get_computed_role(Parameters parameters, JsonValue payload) = 0;
```
Comment (trflynn89) [inline Userland/Libraries/LibWeb/WebDriver/Client.h:78]: Same comment here re: spec order

File: Userland/Services/WebContent/WebDriverClient.ipc [hunk 3]
```diff
@@ -56,4 +56,5 @@ endpoint WebDriverClient {
     take_screenshot() => (Web::WebDriver::Response response)
     take_element_screenshot(DeprecatedString element_id) => (Web::WebDriver::Response response)
     print_page() => (Web::WebDriver::Response response)
+    get_computed_role(DeprecatedString element_id) => (Web::WebDriver::Response response)
```
Comment (trflynn89) [inline Userland/Services/WebContent/WebDriverClient.ipc:59]: Same comment here re: spec order

File: Userland/Services/WebContent/WebDriverConnection.cpp [hunk 4]
```diff
@@ -1805,4 +1805,25 @@ void WebDriverConnection::delete_cookies(Optional<StringView> const& name)
     }
 }
 
+// https://w3c.github.io/webdriver/#dfn-get-computed-role
+Messages::WebDriverClient::GetComputedRoleResponse WebDriverConnection::get_computed_role(DeprecatedString const& element_id)
```
Comment (trflynn89) [inline Userland/Services/WebContent/WebDriverConnection.cpp:1809]: Same comment here re: spec order

File: Userland/Services/WebContent/WebDriverConnection.cpp [hunk 5]
```diff
@@ -1805,4 +1805,25 @@ void WebDriverConnection::delete_cookies(Optional<StringView> const& name)
...
+    // 3. Let element be the result of trying to get a known connected element with url variable element id.
+    auto* element = TRY(get_known_connected_element(element_id));
+
+    // 4. Let role be the result of computing the WAI-ARIA role of element.
+    auto role = element->role_or_default();
+
+    // 5. Return success with data role.
+    JsonObject result;
+    result.set("role"sv, role.view());
+    return result;
```
Comment (trflynn89) [inline Userland/Services/WebContent/WebDriverConnection.cpp:1826]: This should just be:
```c++
return role;
```

If that doesn't compile (idk off hand if `DeprecatedFlyString` is convertible to a `JsonValue`), then:
```c++
return DeprecatedString { role };
```

In general "Return success with <something>" generally means, literally, "return <something>" from here.

File: Userland/Services/WebContent/WebDriverConnection.h [hunk 6]
```diff
@@ -91,6 +91,7 @@ class WebDriverConnection final
     virtual Messages::WebDriverClient::TakeScreenshotResponse take_screenshot() override;
     virtual Messages::WebDriverClient::TakeElementScreenshotResponse take_element_screenshot(DeprecatedString const& element_id) override;
     virtual Messages::WebDriverClient::PrintPageResponse print_page() override;
+    virtual Messages::WebDriverClient::GetComputedRoleResponse get_computed_role(DeprecatedString const& element_id) override;
```
Comment (trflynn89) [inline Userland/Services/WebContent/WebDriverConnection.h:94]: Same comment here re: spec order

File: Userland/Services/WebDriver/Client.cpp [hunk 7]
```diff
@@ -677,4 +677,13 @@ Web::WebDriver::Response Client::print_page(Web::WebDriver::Parameters parameter
     return session->web_content_connection().print_page();
 }
 
+// 12.4.9 https://w3c.github.io/webdriver/#dfn-get-computed-role
+// GET /session/{session id}/element/{element id}/computedrole
+Web::WebDriver::Response Client::get_computed_role(Web::WebDriver::Parameters parameters, AK::JsonValue)
```
Comment (trflynn89) [inline Userland/Services/WebDriver/Client.cpp:682]: Same comment here re: spec order

File: Userland/Services/WebDriver/Client.h [hunk 8]
```diff
@@ -91,6 +91,7 @@ class Client final : public Web::WebDriver::Client {
     virtual Web::WebDriver::Response take_screenshot(Web::WebDriver::Parameters parameters, JsonValue payload) override;
     virtual Web::WebDriver::Response take_element_screenshot(Web::WebDriver::Parameters parameters, JsonValue payload) override;
     virtual Web::WebDriver::Response print_page(Web::WebDriver::Parameters parameters, JsonValue payload) override;
+    virtual Web::WebDriver::Response get_computed_role(Web::WebDriver::Parameters parameters, JsonValue payload) override;
```
Comment (trflynn89) [inline Userland/Services/WebDriver/Client.h:94]: Same comment here re: spec order

File: Userland/Services/WebContent/WebDriverConnection.cpp [hunk 9]
```diff
@@ -1805,4 +1805,25 @@ void WebDriverConnection::delete_cookies(Optional<StringView> const& name)
     }
 }
 
+// https://w3c.github.io/webdriver/#dfn-get-computed-role
```
Comment (trflynn89) [inline Userland/Services/WebContent/WebDriverConnection.cpp:1808]: This line should be;
```c++
// 12.4.9 Get Computed Role, https://w3c.github.io/webdriver/#dfn-get-computed-role
```

I know this is a bit confusing, but the difference between this line and the commented spec links just above this function is that this is a numbered spec API vs something like `delete_cookies` being an "abstract operation".

File: Userland/Libraries/LibWeb/WebDriver/Client.cpp (hunk 1 above)
Comment (Epigenetic) [reply Userland/Libraries/LibWeb/WebDriver/Client.cpp:75]: Gotcha, I had thought it was alphabetical

File: Userland/Services/WebDriver/Client.h (hunk 8 above)
Comment (trflynn89) [inline Userland/Services/WebDriver/Client.h:94]: Almost there, just this one remaining line should be moved just below `is_element_enabled` above

**Review by trflynn89 — APPROVED** [review]

**Pull request merged**